apps/emqx_modules/src/emqx_modules_schema.erl

```diff
@@ -45,6 +46,28 @@ roots() ->
         })
     ].
 
+rewrite_validator(Rules) ->
+    case
+        lists:foldl(
+            fun
+                (#{<<"action">> := subscribe}, Acc) ->
+                    Acc;
+                (#{<<"dest_topic">> := DestTopic} = Rule, InvalidAcc) ->
+                    case catch emqx_topic:validate(name, string:trim(DestTopic)) of
+                        true ->
+                            InvalidAcc;
+                        _ ->
+                            [Rule | InvalidAcc]
+                    end
```

Comment:
Please consider direct `catch` deprecated and we should never use it for new code in EMQX codebase.
```suggestion
                    try
                        true = emqx_topic:validate(name, DestTopic),
                        InvalidAcc
                    catch _:_ ->
                        [DestTopic | InvalidAcc]
                    end
```

Reply:
Ack